File: crates/zugfolge-conductor/src/interior_types.rs
use serde::{Deserialize, Serialize};
use zugfolge_demand::{
    ComfortClassV1, DemandEvaluationV1, SpaceNeedsV1, TrainCapacityV1, TrainServiceV1,
};
use zugfolge_fleet::{
    MobilizationSnapshot, VehicleConfigurationV1, release_catalog::FleetAuthorityRelease,
};

use crate::{
    ConductorPassengerBindingV1, InteriorPlaceKindV1, PassengerActivityV1, PassengerPostureV1,
    PassengerProjectionPhaseV1,
};

pub const LAYOUT_INPUT_SCHEMA: &str = "conductor-interior-layout-input/v1";
pub const LAYOUT_SCHEMA: &str = "interior-layout/v1";
pub const GEOMETRY_POLICY_SCHEMA: &str = "conductor-interior-geometry-policy/v1";
pub const INTERIOR_V2_SCHEMA: &str = "interior-passenger-places/v2";
pub const PROJECTION_V2_SCHEMA: &str = "passenger-projection/v2";
pub const PROJECTION_INPUT_V2_SCHEMA: &str = "conductor-passenger-projection-input/v2";

#[derive(Clone, Copy, Debug, Eq, PartialEq, Ord, PartialOrd, Serialize, Deserialize)]
#[serde(rename_all = "snake_case")]
pub enum InteriorDeckIdV1 {
    Main,
    Lower,
    Upper,
}

#[derive(Clone, Debug, Eq, PartialEq, Serialize, Deserialize)]
#[serde(rename_all = "camelCase", deny_unknown_fields)]
pub struct InteriorPointV1 {
    pub vehicle_id: String,
    pub body_id: String,
    pub deck_id: InteriorDeckIdV1,
    pub x_mm: i64,
    pub y_mm: i64,
}

#[derive(Clone, Debug, Eq, PartialEq, Serialize, Deserialize)]
#[serde(rename_all = "camelCase", deny_unknown_fields)]
pub struct InteriorRectV1 {
    pub x_mm: i64,
    pub y_mm: i64,
    pub length_mm: i64,
    pub width_mm: i64,
}

#[derive(Clone, Debug, Eq, PartialEq, Serialize, Deserialize)]
#[serde(rename_all = "camelCase", deny_unknown_fields)]
pub struct InteriorLayoutBindingV1 {
    pub world_id: String,
    pub period_id: String,
    pub operator_id: String,
    pub formation_id: String,
    pub formation_revision: u64,
    pub fleet_state_hash: String,
    pub fleet_authority_release_id: String,
    pub fleet_authority_release_hash: String,
    pub mobilization_snapshot_hash: String,
    pub geometry_policy_hash: String,
    pub art_release_id: String,
    pub art_manifest_hash: String,
}

#[derive(Clone, Debug, Eq, PartialEq, Serialize, Deserialize)]
#[serde(rename_all = "camelCase", deny_unknown_fields)]
pub struct InteriorStairGeometryV1 {
    pub stair_id: String,
    pub from_deck_id: InteriorDeckIdV1,
    pub to_deck_id: InteriorDeckIdV1,
    pub at_mm: u32,
}

#[derive(Clone, Debug, Eq, PartialEq, Serialize, Deserialize)]
#[serde(rename_all = "camelCase", deny_unknown_fields)]
pub struct InteriorBodyGeometryV1 {
    pub body_id: String,
    pub length_mm: u32,
    pub width_mm: u32,
    pub deck_ids: Vec<InteriorDeckIdV1>,
    pub entrance_deck_id: InteriorDeckIdV1,
    pub door_positions_mm: Vec<u32>,
    pub stairs: Vec<InteriorStairGeometryV1>,
    pub gap_after_mm: u32,
    pub front_gangway: bool,
    pub rear_gangway: bool,
}

#[derive(Clone, Debug, Eq, PartialEq, Serialize, Deserialize)]
#[serde(rename_all = "camelCase", deny_unknown_fields)]
pub struct InteriorVehicleGeometryV1 {
    pub vehicle_type_id: u64,
    pub configuration_hash: Option<String>,
    pub art_family: String,
    pub bodies: Vec<InteriorBodyGeometryV1>,
}

#[derive(Clone, Debug, Eq, PartialEq, Serialize, Deserialize)]
#[serde(rename_all = "camelCase", deny_unknown_fields)]
pub struct InteriorGeometryPolicyV1 {
    pub schema_version: String,
    pub policy_id: String,
    pub vehicle_types: Vec<InteriorVehicleGeometryV1>,
}

#[derive(Clone, Debug, Eq, PartialEq, Serialize, Deserialize)]
#[serde(rename_all = "camelCase", deny_unknown_fields)]
pub struct BuildInteriorLayoutInputV1 {
    pub schema_version: String,
    pub binding: InteriorLayoutBindingV1,
    pub authority_release: FleetAuthorityRelease,
    pub mobilization: MobilizationSnapshot,
    pub geometry_policy: InteriorGeometryPolicyV1,
}

#[derive(Clone, Debug, Eq, PartialEq, Serialize, Deserialize)]
#[serde(rename_all = "camelCase", deny_unknown_fields)]
pub struct InteriorPassengerPlaceV2 {
    pub place_id: String,
    pub vehicle_id: String,
    pub body_id: String,
    pub deck_id: InteriorDeckIdV1,
    pub x_mm: i64,
    pub y_mm: i64,
    pub comfort_class: ComfortClassV1,
    pub kind: InteriorPlaceKindV1,
    pub space_needs: Vec<SpaceNeedsV1>,
}

/// Zusätzliche exklusive Raumressource; zählt niemals als weiterer Fahrgastplatz.
#[derive(Clone, Debug, Eq, PartialEq, Serialize, Deserialize)]
#[serde(rename_all = "camelCase", deny_unknown_fields)]
pub struct InteriorSpecialBayV1 {
    pub space_id: String,
    pub vehicle_id: String,
    pub body_id: String,
    pub deck_id: InteriorDeckIdV1,
    pub x_mm: i64,
    pub y_mm: i64,
    pub space_need: SpaceNeedsV1,
}

#[derive(Clone, Debug, Eq, PartialEq, Serialize, Deserialize)]
#[serde(rename_all = "camelCase", deny_unknown_fields)]
pub struct InteriorPassengerPlacesV2 {
    pub schema_version: String,
    pub world_id: String,
    pub train_run_id: String,
    pub layout_id: String,
    pub source_layout_hash: String,
    pub layout_hash: String,
    pub places: Vec<InteriorPassengerPlaceV2>,
    pub special_bays: Vec<InteriorSpecialBayV1>,
}

#[derive(Clone, Debug, Eq, PartialEq, Serialize, Deserialize)]
#[serde(rename_all = "camelCase", deny_unknown_fields)]
pub struct VisiblePassengerV2 {
    pub passenger_key: String,
    pub place_id: String,
    pub space_id: Option<String>,
    pub vehicle_id: String,
    pub body_id: String,
    pub deck_id: InteriorDeckIdV1,
    pub x_mm: i64,
    pub y_mm: i64,
    pub comfort_class: ComfortClassV1,
    pub space_needs: SpaceNeedsV1,
    pub posture: PassengerPostureV1,
    pub appearance_variant: u32,
    pub activity: PassengerActivityV1,
}

#[derive(Clone, Debug, Eq, PartialEq, Serialize, Deserialize)]
#[serde(rename_all = "camelCase", deny_unknown_fields)]
pub struct PassengerProjectionV2 {
    pub schema_version: String,
    pub binding: ConductorPassengerBindingV1,
    pub segment_id: String,
    pub from_stop_id: String,
    pub to_stop_id: String,
    pub phase: PassengerProjectionPhaseV1,
    pub current_stop_id: Option<String>,
    pub layout_id: String,
    pub source_layout_hash: String,
    pub layout_hash: String,
    pub as_of_ms: i64,
    pub passengers: Vec<VisiblePassengerV2>,
    pub state_hash: String,
}

#[derive(Clone, Debug, Eq, PartialEq, Serialize, Deserialize)]
#[serde(rename_all = "camelCase", deny_unknown_fields)]
pub struct ProjectConductorPassengersInputV2 {
    pub schema_version: String,
    pub binding: ConductorPassengerBindingV1,
    pub evaluation: DemandEvaluationV1,
    pub service: TrainServiceV1,
    pub interior: InteriorPassengerPlacesV2,
    #[serde(default, skip_serializing_if = "Option::is_none")]
    pub previous_projection: Option<PassengerProjectionV2>,
}

#[derive(Clone, Copy, Debug, Eq, PartialEq, Serialize, Deserialize)]
#[serde(rename_all = "snake_case")]
pub enum InteriorObstacleKindV1 {
    Wall,
    Cab,
    Seat,
    Toilet,
    AccessibleToilet,
    Stair,
    Bicycle,
    Stroller,
    Wheelchair,
}

#[derive(Clone, Debug, Eq, PartialEq, Serialize, Deserialize)]
#[serde(rename_all = "camelCase", deny_unknown_fields)]
pub struct InteriorObstacleV1 {
    pub obstacle_id: String,
    pub vehicle_id: String,
    pub body_id: String,
    pub deck_id: InteriorDeckIdV1,
    pub kind: InteriorObstacleKindV1,
    pub rect: InteriorRectV1,
}

#[derive(Clone, Debug, Eq, PartialEq, Serialize, Deserialize)]
#[serde(rename_all = "camelCase", deny_unknown_fields)]
pub struct InteriorNodeV1 {
    pub node_id: String,
    pub point: InteriorPointV1,
}

#[derive(Clone, Copy, Debug, Eq, PartialEq, Serialize, Deserialize)]
#[serde(rename_all = "snake_case")]
pub enum InteriorEdgeKindV1 {
    Walk,
    Stair,
    Gangway,
}

#[derive(Clone, Debug, Eq, PartialEq, Serialize, Deserialize)]
#[serde(rename_all = "camelCase", deny_unknown_fields)]
pub struct InteriorEdgeV1 {
    pub edge_id: String,
    pub from_node_id: String,
    pub to_node_id: String,
    pub kind: InteriorEdgeKindV1,
    pub length_mm: u64,
    pub wheelchair_accessible: bool,
}

#[derive(Clone, Copy, Debug, Eq, PartialEq, Serialize, Deserialize)]
#[serde(rename_all = "snake_case")]
pub enum InteriorInteractionKindV1 {
    Passenger,
    Door,
    Toilet,
    AccessibleToilet,
    Cab,
    Stair,
    Bicycle,
    Stroller,
    Wheelchair,
}

#[derive(Clone, Debug, Eq, PartialEq, Serialize, Deserialize)]
#[serde(rename_all = "camelCase", deny_unknown_fields)]
pub struct InteriorInteractionV1 {
    pub interaction_id: String,
    pub kind: InteriorInteractionKindV1,
    pub target_id: String,
    pub node_id: String,
}

#[derive(Clone, Debug, Eq, PartialEq, Serialize, Deserialize)]
#[serde(rename_all = "camelCase", deny_unknown_fields)]
pub struct InteriorDoorV1 {
    pub door_id: String,
    pub vehicle_id: String,
    pub body_id: String,
    pub deck_id: InteriorDeckIdV1,
    pub side: String,
    pub rect: InteriorRectV1,
    pub node_id: String,
}

#[derive(Clone, Copy, Debug, Eq, PartialEq, Serialize, Deserialize)]
#[serde(rename_all = "snake_case")]
pub enum InteriorSeatFacingV1 {
    Forward,
    Backward,
}

#[derive(Clone, Debug, Eq, PartialEq, Serialize, Deserialize)]
#[serde(rename_all = "camelCase", deny_unknown_fields)]
pub struct InteriorSeatV1 {
    pub place_id: String,
    pub obstacle_id: String,
    pub facing: InteriorSeatFacingV1,
}

#[derive(Clone, Debug, Eq, PartialEq, Serialize, Deserialize)]
#[serde(rename_all = "camelCase", deny_unknown_fields)]
pub struct InteriorBodyV1 {
    pub body_id: String,
    pub vehicle_id: String,
    pub length_mm: u32,
    pub width_mm: u32,
    pub vehicle_offset_mm: u32,
    pub formation_offset_mm: u64,
    pub reversed: bool,
    pub deck_ids: Vec<InteriorDeckIdV1>,
    pub entrance_deck_id: InteriorDeckIdV1,
    pub passenger_accessible: bool,
    pub front_gangway: bool,
    pub rear_gangway: bool,
    pub gap_after_mm: u32,
}

#[derive(Clone, Debug, Eq, PartialEq, Serialize, Deserialize)]
#[serde(rename_all = "camelCase", deny_unknown_fields)]
pub struct InteriorVehicleV1 {
    pub vehicle_id: String,
    pub vehicle_type_id: u64,
    pub configuration: Option<VehicleConfigurationV1>,
    pub configuration_hash: Option<String>,
    pub art_family: String,
    pub capacity: TrainCapacityV1,
    pub bodies: Vec<InteriorBodyV1>,
}

#[derive(Clone, Debug, Eq, PartialEq, Serialize, Deserialize)]
#[serde(rename_all = "camelCase", deny_unknown_fields)]
pub struct InteriorLayoutV1 {
    pub schema_version: String,
    pub binding: InteriorLayoutBindingV1,
    pub layout_id: String,
    pub layout_hash: String,
    pub capacity: TrainCapacityV1,
    pub vehicles: Vec<InteriorVehicleV1>,
    pub passenger_places: Vec<InteriorPassengerPlaceV2>,
    pub special_bays: Vec<InteriorSpecialBayV1>,
    pub obstacles: Vec<InteriorObstacleV1>,
    pub nodes: Vec<InteriorNodeV1>,
    pub edges: Vec<InteriorEdgeV1>,
    pub interactions: Vec<InteriorInteractionV1>,
    pub doors: Vec<InteriorDoorV1>,
    pub seats: Vec<InteriorSeatV1>,
    pub entrance_node_id: String,
}

#[derive(Clone, Debug, Eq, PartialEq, Serialize, Deserialize)]
#[serde(rename_all = "camelCase", deny_unknown_fields)]
pub struct InteriorLayoutIssueV1 {
    pub code: String,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub vehicle_id: Option<String>,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub body_id: Option<String>,
}

#[derive(Clone, Debug, Eq, PartialEq, Serialize, Deserialize)]
#[serde(rename_all = "camelCase", deny_unknown_fields)]
pub struct BindInteriorPassengerPlacesInputV1 {
    pub schema_version: String,
    pub layout: InteriorLayoutV1,
    pub train_run_id: String,
    pub service: TrainServiceV1,
}

#[derive(Clone, Debug, Eq, PartialEq, Serialize, Deserialize)]
#[serde(rename_all = "camelCase", deny_unknown_fields)]
pub struct FindInteriorPathInputV1 {
    pub schema_version: String,
    pub layout: InteriorLayoutV1,
    pub expected_layout_hash: String,
    pub from_node_id: String,
    pub to_node_id: String,
    pub wheelchair: bool,
}

#[derive(Clone, Debug, Eq, PartialEq, Serialize, Deserialize)]
#[serde(rename_all = "camelCase", deny_unknown_fields)]
pub struct InteriorPathV1 {
    pub schema_version: String,
    pub layout_hash: String,
    pub node_ids: Vec<String>,
    pub edge_ids: Vec<String>,
    pub length_mm: u64,
}

#[derive(Clone, Debug, Eq, PartialEq, Serialize, Deserialize)]
#[serde(rename_all = "camelCase", deny_unknown_fields)]
pub struct CheckInteriorMovementInputV1 {
    pub schema_version: String,
    pub layout: InteriorLayoutV1,
    pub expected_layout_hash: String,
    pub from: InteriorPointV1,
    pub to: InteriorPointV1,
    pub transition_edge_id: Option<String>,
    pub wheelchair: bool,
}

#[derive(Clone, Debug, Eq, PartialEq, Serialize, Deserialize)]
#[serde(rename_all = "camelCase", deny_unknown_fields)]
pub struct InteriorMovementResultV1 {
    pub schema_version: String,
    pub layout_hash: String,
    pub allowed: bool,
    pub issue: Option<String>,
}

File: packages/runtime-native/src/interior-types.ts
import type { FleetAuthorityRelease, NativeFleetMobilizationSnapshot } from "./index.js";
import type { ConductorPassengerBindingV1, PassengerComfortClass, PassengerProjectionV1, PassengerSpaceNeeds } from "./conductor.js";
import type { M5VehicleConfigurationV1 } from "./vehicle-configuration.js";

export type InteriorDeckId = "main" | "lower" | "upper";
export interface InteriorPointV1 {
  readonly vehicleId: string; readonly bodyId: string; readonly deckId: InteriorDeckId;
  readonly xMm: number; readonly yMm: number;
}
export interface InteriorRectV1 { readonly xMm: number; readonly yMm: number; readonly lengthMm: number; readonly widthMm: number }
export interface InteriorLayoutBindingV1 {
  readonly worldId: string; readonly periodId: string; readonly operatorId: string;
  readonly formationId: string; readonly formationRevision: number; readonly fleetStateHash: string;
  readonly fleetAuthorityReleaseId: string; readonly fleetAuthorityReleaseHash: string;
  readonly mobilizationSnapshotHash: string; readonly geometryPolicyHash: string;
  readonly artReleaseId: string; readonly artManifestHash: string;
}
export interface InteriorStairGeometryV1 {
  readonly stairId: string; readonly fromDeckId: InteriorDeckId; readonly toDeckId: InteriorDeckId; readonly atMm: number;
}
export interface InteriorBodyGeometryV1 {
  readonly bodyId: string; readonly lengthMm: number; readonly widthMm: number;
  readonly deckIds: readonly InteriorDeckId[]; readonly entranceDeckId: InteriorDeckId;
  readonly doorPositionsMm: readonly number[]; readonly stairs: readonly InteriorStairGeometryV1[];
  readonly gapAfterMm: number; readonly frontGangway: boolean; readonly rearGangway: boolean;
}
export interface InteriorVehicleGeometryV1 {
  readonly vehicleTypeId: number; readonly configurationHash: string | null; readonly artFamily: string;
  readonly bodies: readonly InteriorBodyGeometryV1[];
}
export interface InteriorGeometryPolicyV1 {
  readonly schemaVersion: "conductor-interior-geometry-policy/v1";
  readonly policyId: string; readonly vehicleTypes: readonly InteriorVehicleGeometryV1[];
}
export interface BuildInteriorLayoutInputV1 {
  readonly schemaVersion: "conductor-interior-layout-input/v1";
  readonly binding: InteriorLayoutBindingV1; readonly authorityRelease: FleetAuthorityRelease;
  readonly mobilization: NativeFleetMobilizationSnapshot; readonly geometryPolicy: InteriorGeometryPolicyV1;
}
export interface InteriorCapacityV1 {
  readonly standardSeats: number; readonly standardStanding: number; readonly premiumSeats: number;
  readonly wheelchairSpaces: number; readonly bicycleSpaces: number; readonly strollerSpaces: number;
}
export interface InteriorPassengerPlaceV2 extends InteriorPointV1 {
  readonly placeId: string; readonly comfortClass: PassengerComfortClass;
  readonly kind: "seat" | "standing"; readonly spaceNeeds: readonly PassengerSpaceNeeds[];
}
export interface InteriorSpecialBayV1 extends InteriorPointV1 { readonly spaceId: string; readonly spaceNeed: PassengerSpaceNeeds }
export interface InteriorPassengerPlacesV2 {
  readonly schemaVersion: "interior-passenger-places/v2";
  readonly worldId: string; readonly trainRunId: string; readonly layoutId: string;
  readonly sourceLayoutHash: string; readonly layoutHash: string;
  readonly places: readonly InteriorPassengerPlaceV2[]; readonly specialBays: readonly InteriorSpecialBayV1[];
}
export interface VisiblePassengerV2 extends InteriorPointV1 {
  readonly passengerKey: string; readonly placeId: string; readonly spaceId: string | null;
  readonly comfortClass: PassengerComfortClass; readonly spaceNeeds: PassengerSpaceNeeds;
  readonly posture: "seated" | "standing"; readonly appearanceVariant: number; readonly activity: "onboard" | "alighting";
}
export interface PassengerProjectionV2 extends Omit<PassengerProjectionV1, "schemaVersion" | "passengers"> {
  readonly schemaVersion: "passenger-projection/v2"; readonly sourceLayoutHash: string;
  readonly passengers: readonly VisiblePassengerV2[];
}
export interface ProjectConductorPassengersInputV2 {
  readonly schemaVersion: "conductor-passenger-projection-input/v2"; readonly binding: ConductorPassengerBindingV1;
  readonly evaluation: Readonly<Record<string, unknown>>; readonly service: Readonly<Record<string, unknown>>;
  readonly interior: InteriorPassengerPlacesV2; readonly previousProjection?: PassengerProjectionV2;
}
export interface InteriorBodyV1 {
  readonly bodyId: string; readonly vehicleId: string; readonly lengthMm: number; readonly widthMm: number;
  readonly vehicleOffsetMm: number; readonly formationOffsetMm: number; readonly reversed: boolean;
  readonly deckIds: readonly InteriorDeckId[]; readonly entranceDeckId: InteriorDeckId;
  readonly passengerAccessible: boolean; readonly frontGangway: boolean; readonly rearGangway: boolean; readonly gapAfterMm: number;
}
export interface InteriorVehicleV1 {
  readonly vehicleId: string; readonly vehicleTypeId: number; readonly configuration: M5VehicleConfigurationV1 | null;
  readonly configurationHash: string | null; readonly artFamily: string;
  readonly capacity: InteriorCapacityV1; readonly bodies: readonly InteriorBodyV1[];
}
export interface InteriorObstacleV1 extends Omit<InteriorPointV1, "xMm" | "yMm"> {
  readonly obstacleId: string;
  readonly kind: "wall" | "cab" | "seat" | "toilet" | "accessible_toilet" | "stair" | "bicycle" | "stroller" | "wheelchair";
  readonly rect: InteriorRectV1;
}
export interface InteriorNodeV1 { readonly nodeId: string; readonly point: InteriorPointV1 }
export interface InteriorEdgeV1 {
  readonly edgeId: string; readonly fromNodeId: string; readonly toNodeId: string;
  readonly kind: "walk" | "stair" | "gangway"; readonly lengthMm: number; readonly wheelchairAccessible: boolean;
}
export interface InteriorInteractionV1 {
  readonly interactionId: string; readonly kind: "passenger" | "door" | "toilet" | "accessible_toilet" | "cab" | "stair" | "bicycle" | "stroller" | "wheelchair";
  readonly targetId: string; readonly nodeId: string;
}
export interface InteriorDoorV1 extends Omit<InteriorPointV1, "xMm" | "yMm"> {
  readonly doorId: string; readonly side: "left" | "right"; readonly rect: InteriorRectV1; readonly nodeId: string;
}
export interface InteriorSeatV1 { readonly placeId: string; readonly obstacleId: string; readonly facing: "forward" | "backward" }
export interface InteriorLayoutV1 {
  readonly schemaVersion: "interior-layout/v1"; readonly binding: InteriorLayoutBindingV1;
  readonly layoutId: string; readonly layoutHash: string; readonly capacity: InteriorCapacityV1;
  readonly vehicles: readonly InteriorVehicleV1[]; readonly passengerPlaces: readonly InteriorPassengerPlaceV2[];
  readonly specialBays: readonly InteriorSpecialBayV1[]; readonly obstacles: readonly InteriorObstacleV1[];
  readonly nodes: readonly InteriorNodeV1[]; readonly edges: readonly InteriorEdgeV1[];
  readonly interactions: readonly InteriorInteractionV1[]; readonly doors: readonly InteriorDoorV1[];
  readonly seats: readonly InteriorSeatV1[]; readonly entranceNodeId: string;
}
export interface BindInteriorPassengerPlacesInputV1 {
  readonly schemaVersion: "conductor-interior-bind-input/v1"; readonly layout: InteriorLayoutV1;
  readonly trainRunId: string; readonly service: Readonly<Record<string, unknown>>;
}
export interface FindInteriorPathInputV1 {
  readonly schemaVersion: "conductor-interior-path-input/v1"; readonly layout: InteriorLayoutV1;
  readonly expectedLayoutHash: string; readonly fromNodeId: string; readonly toNodeId: string; readonly wheelchair: boolean;
}
export interface InteriorPathV1 {
  readonly schemaVersion: "interior-path/v1"; readonly layoutHash: string;
  readonly nodeIds: readonly string[]; readonly edgeIds: readonly string[]; readonly lengthMm: number;
}
export interface CheckInteriorMovementInputV1 {
  readonly schemaVersion: "conductor-interior-movement-input/v1"; readonly layout: InteriorLayoutV1;
  readonly expectedLayoutHash: string; readonly from: InteriorPointV1; readonly to: InteriorPointV1;
  readonly transitionEdgeId: string | null; readonly wheelchair: boolean;
}
export interface InteriorMovementResultV1 {
  readonly schemaVersion: "interior-movement-result/v1"; readonly layoutHash: string;
  readonly allowed: boolean; readonly issue: string | null;
}
